assets/src/block-editor/TermSelector/index.js

```diff
@@ -0,0 +1,369 @@
+// This component is based on the one used by WordPress for categories (without the hierarchical aspect, since we don't need it):
+// https://github.com/WordPress/gutenberg/blob/trunk/packages/editor/src/components/post-taxonomies/hierarchical-term-selector.js
+
+/**
+ * WordPress dependencies
+ */
+import {useMemo, useState} from '@wordpress/element';
+import {
+  Button,
+  CheckboxControl,
+  TextControl,
+  withFilters,
+} from '@wordpress/components';
+import {useDispatch, useSelect} from '@wordpress/data';
+import {useDebounce} from '@wordpress/compose';
+import {store as coreStore} from '@wordpress/core-data';
+import {decodeEntities} from '@wordpress/html-entities';
+
+const {__, _n, _x, sprintf} = wp.i18n;
+const {speak} = wp.a11y;
+
+/**
+ * Module Constants
+ */
+const DEFAULT_QUERY = {
+  per_page: -1,
+  orderby: 'name',
+  order: 'asc',
+  _fields: 'id,name',
+  context: 'view',
+};
+
+const MIN_TERMS_COUNT_FOR_FILTER = 8;
+
+const EMPTY_ARRAY = [];
+
+/**
+ * Sort Terms by Selected.
+ *
+ * @param {Object[]} termsTree Array of terms in tree format.
+ * @param {number[]} terms     Selected terms.
+ *
+ * @return {Object[]} Sorted array of terms.
+ */
+export function sortBySelected(termsTree, terms) {
+  const treeHasSelection = termTree => {
+    return terms.indexOf(termTree.id) !== -1;
+  };
+
+  const termIsSelected = (termA, termB) => {
+    const termASelected = treeHasSelection(termA);
+    const termBSelected = treeHasSelection(termB);
+
+    if (termASelected === termBSelected) {
```

Comment:
I don't see necesary this return since we're already returning 0 when `termASelected` or `termBSelected` are falsy.